Tests/AppTests/SearchFilterTests.swift

```diff
@@ -0,0 +1,163 @@
+@testable import App
+
+import SQLKit
+import XCTVapor
+import SnapshotTesting
+
+class SearchFilterTests: AppTestCase {
+    
+    func test_allSearchFilters() {
+        XCTAssertEqual(
+            SearchFilterParser
+                .allSearchFilters
+                .map { $0.key }
+                .sorted(),
+            [ "license", "stars", "updated" ]
+        )
+    }
+    
+    func test_parseTerm() {
+        let parser = SearchFilterParser()
+        
+        func mockParse(term: String) throws -> MockSearchFilter {
+            try XCTUnwrap(parser.parse(term: term, allFilters: [
+                MockSearchFilter.self
+            ]) as? MockSearchFilter)
+        }
+        
+        do { // No colon
+            XCTAssertNil(parser.parse(term: "a"))
+        }
+        
+        do { // Too many colons
+            XCTAssertNil(parser.parse(term: "a:b:c"))
+        }
+        
+        do { // Comparison method
+            try XCTAssertEqual(mockParse(term: "mock:1").comparison, .match)
+            try XCTAssertEqual(mockParse(term: "mock:>1").comparison, .greaterThan)
+            try XCTAssertEqual(mockParse(term: "mock:<1").comparison, .lessThan)
+            try XCTAssertEqual(mockParse(term: "mock:!1").comparison, .negativeMatch)
+        }
+        
+        do { // Correct value
+            try XCTAssertEqual(mockParse(term: "mock:test").value, "test")
+            try XCTAssertEqual(mockParse(term: "mock:!test").value, "test")
+            
+            // terms are usually tokenised based on spaces meaning this should, in theory,
+            // never happen. However, the filter system does support it.
+            try XCTAssertEqual(mockParse(term: "mock:!with space").value, "with space")
```

Comment:
This was more just a test of my own sanity to see what would happen if a space was passed through to the grammar parser. It treats everything after the colon as one string, regardless of spaces.

However, as I tried to explain in the comment, this will never actually happen since we separate on spaces right at the beginning before it's passed to the parser.

You're right in saying that we should support something like `mock:"with space"` but that's when we look to enrich the grammar with more features. Fortunately, once we support the tokenisation part - this test provides that the parser itself handles it fine 👍